pkgs/development/python-modules/wgpu-py/default.nix

```diff
@@ -0,0 +1,157 @@
+{
+  lib,
+  stdenv,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  hatchling,
+
+  # preBuild
+  wgpu-native,
+
+  # dependencies
+  cffi,
+  rubicon-objc,
+  sniffio,
+
+  # optional dependency
+  glfw,
+
+  # docs
+  sphinx-rtd-theme,
+  sphinxHook,
+
+  # tests
+  anyio,
+  imageio,
+  numpy,
+  psutil,
+  pytest,
+  ruff,
+  trio,
+
+  # passthru
+  wgpu-py,
+  testers,
+  nix-update-script,
+}:
+buildPythonPackage rec {
+  pname = "wgpu-py";
+  version = "0.21.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "pygfx";
+    repo = pname;
+    tag = "v${version}";
+    hash = "sha256-XlV0ovIF3w/u6f65+3c4zAfisCoQDbzMiINJJTR/I6o=";
+  };
+
+  # wgpu-py expects to have an appropriately named wgpu-native library in wgpu/resources
+  preBuild = ''
+    cp ${wgpu-native}/lib/libwgpu_native${stdenv.hostPlatform.extensions.library} \
+      wgpu/resources/libwgpu_native-release${stdenv.hostPlatform.extensions.library}
```

Comment:
I wonder if this should be in `preBuild`...
Anyway, I think that `ln -s` would be better.